.github/workflows/cypress-tests-runner.yml

```
@@ -261,3 +254,207 @@ jobs:
+
+      - name: Install sccache
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: taiki-e/install-action@v2.41.10
+        with:
+          tool: sccache
+          checksum: true
+
+      - name: Install Diesel CLI
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: baptiste0928/cargo-install@v3.1.1
+        with:
+          crate: diesel_cli
+          features: postgres
+          args: --no-default-features
+
+      - name: Install grcov
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: taiki-e/install-action@v2.41.10
+        with:
+          tool: grcov
+          checksum: true
+
+      - name: Install Just
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: taiki-e/install-action@v2.41.10
+        with:
+          tool: just
+          checksum: true
+
+      - name: Install Node.js
+        if: ${{ env.RUN_TESTS == 'true' }}
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install Cypress and dependencies
+        if: ${{ env.RUN_TESTS == 'true' }}
+        run: |
+          npm ci --prefix ./cypress-tests-v2
+
+      - name: Run database migrations
+        if: ${{ env.RUN_TESTS == 'true' }}
+        shell: bash
+        env:
+          DATABASE_URL: postgres://db_user:db_pass@localhost:5432/hyperswitch_db
+        run: just migrate_v2 run --locked-schema
+
+      - name: Insert card info into the database
+        if: ${{ env.RUN_TESTS == 'true' }}
+        run: |
+          PGPASSWORD=db_pass psql --host=localhost --port=5432 --username=db_user --dbname=hyperswitch_db --command "\copy cards_info FROM '.github/data/cards_info.csv' DELIMITER ',' CSV HEADER;"
+
+      - name: Build project
+        if: ${{ env.RUN_TESTS == 'true' }}
+        env:
+          RUSTFLAGS: "-Cinstrument-coverage"
+        run: just build_v2 --jobs 3
+
+      - name: Setup Local Server
+        if: ${{ env.RUN_TESTS == 'true' }}
+        env:
+          LLVM_PROFILE_FILE: "coverage.profraw"
+        run: |
+          # Start the server in the background
+          target/debug/router &
+
+          SERVER_PID=$!
+          echo "PID=${SERVER_PID}" >> $GITHUB_ENV
+
+          # Wait for the server to start in port 8080
+          COUNT=0
+          while ! nc -z localhost 8080; do
+            if [ $COUNT -gt 12 ]; then # Wait for up to 2 minutes (12 * 10 seconds)
+              echo "Server did not start within a reasonable time. Exiting."
+              kill ${SERVER_PID}
+              exit 1
+            else
+              COUNT=$((COUNT+1))
+              sleep 10
+            fi
+          done
+
+      - name: Run Cypress tests
+        if: ${{ env.RUN_TESTS == 'true' }}
+        env:
+          CYPRESS_BASEURL: "http://localhost:8080"
+          ROUTER__SERVER__WORKERS: 4
+        shell: bash -leuo pipefail {0}
+        continue-on-error: true
+        run: |
+          scripts/execute_cypress.sh "" "" "cypress-tests-v2"
```

Nit:

```suggestion
        run: scripts/execute_cypress.sh "" "" "cypress-tests-v2"
```

Can also add a comment clarifying why the first 2 args are empty. It's not evident unless someone checks the source of the script file.

---

Done